expr/consolidations/consolidations.go

```diff
@@ -119,36 +130,53 @@ func SummarizeValues(f string, values []float64) float64 {
 	switch f {
 	case "sum", "total":
 		for _, av := range values {
-			rv += av
+			if !math.IsNaN(rv) {
+				rv += av
+				total++
+			}
 		}
 
 	case "avg", "average", "avg_zero":
 		for _, av := range values {
-			rv += av
+			if !math.IsNaN(rv) {
+				rv += av
+				total++
+			}
+		}
+		if total == 0 {
+			return math.NaN()
 		}
-		rv /= float64(len(values))
+		rv = rv / float64(total)
```

Comment:
Maybe it should be `rv /= float64(total)`? Just for consistency reasons.

Reply:
makes total sense.

There are more typos `IsNaN(rv)` instead of `IsNaN(av)` in the code.